target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-rtl30vw.dts

```diff
@@ -0,0 +1,407 @@
+			label = "tp27";
+			default-state = "keep";
+		};
+
+		led8 {
+			gpios = <&led_gpio 8 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:wlan2g";
+			default-state = "off";
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led9 {
+			gpios = <&led_gpio 9 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:wlan5g";
+			default-state = "off";
+			linux,default-trigger = "phy1tpt";
+		};
+
+		led10 {
+			gpios = <&led_gpio 10 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:wps";
+			default-state = "off";
+		};
+
+		led11 {
+			gpios = <&led_gpio 11 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:voip";
+			default-state = "off";
+		};
+
+		led12 {
+			gpios = <&led_gpio 12 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s1";
+			default-state = "off";
+		};
+
+		led13 {
+			gpios = <&led_gpio 13 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s2";
+			default-state = "off";
+		};
+
+		led14 {
+			gpios = <&led_gpio 14 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s3";
+			default-state = "off";
+		};
+
+		led15 {
+			gpios = <&led_gpio 15 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s4";
+			default-state = "off";
+		};
+
+		led50 {
+			gpios = <&tlmm 3 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:red:signal";
+			default-state = "off";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		button@1 {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&tlmm 5 GPIO_ACTIVE_LOW>;
+		};
+
+		button@2 {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&tlmm 63 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	soc {
+		tcsr@194b000 {
+			/* select hostmode */
+			compatible = "qcom,tcsr";
+			reg = <0x194b000 0x100>;
+			qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+			status = "ok";
+		};
+
+		ess_tcsr@1953000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1953000 0x1000>;
+			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+		};
+
+		tcsr@1949000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1949000 0x100>;
+			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+		};
+
+		tcsr@1957000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1957000 0x100>;
+			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+		};
+
+		rng@22000 {
+			status = "ok";
+		};
+
+		crypto@8e3a000 {
+			status = "ok";
+		};
+
+		watchdog@b017000 {
+			status = "ok";
+		};
+
+		mdio@90000 {
+			status = "okay";
+		};
+
+		ess-switch@c000000 {
+			status = "okay";
+		};
+
+		ess-psgmii@98000 {
+			status = "okay";
+		};
+
+		edma@c080000 {
+			status = "okay";
+		};
+
+		usb3_ss_phy: ssphy@9a000 {
+			status = "ok";
+		};
+
+		usb3_hs_phy: hsphy@a6000 {
+			status = "ok";
+		};
+
+		usb3: usb3@8af8800 {
+			status = "ok";
+		};
+
+		usb2_hs_phy: hsphy@a8000 {
+			status = "ok";
+		};
+
+		usb2: usb2@60f8800 {
+			status = "ok";
+		};
+	};
+};
+
+&blsp_dma {
+	status = "ok";
+};
+
+
+&blsp1_spi1 {
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	status = "ok";
+	cs-gpios = <&tlmm 54 0>, <&tlmm 59 0>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor", "n25q128a11";
```

Comment:
What is "n25q128a11" needed for?

Reply:
```"n25q128a11"``` is for askey u-boot. It mangle in dts.

Reply:
Why?
All it can do is complain with warning after FDT is loaded as partitions are hardcoded.